core/templates/services/literally-canvas-helper.service.ts

```diff
@@ -64,10 +96,22 @@ angular.module('oppia').factory('LiterallyCanvasHelperService', [
           y += 0.5;
         }
         id = 'rectangle-' + shape.id;
-        return ('<rect id="' + id + '" x="' + x + '" y="' + y + '" width="' +
-        width + '" height="' + height + '" stroke="' + shape.strokeColor +
-        '" fill="' + shape.fillColor + '" stroke-width="' + shape.strokeWidth +
-        '" />');
+        var rect = document.createElement('rect');
+        var attributes = {
+          id: id,
+          x: x,
+          y: y,
+          width: width,
+          height: height,
+          stroke: shape.strokeColor,
+          fill: shape.fillColor,
+        };
+        for (var attrName in attributes) {
+          rect.setAttribute(attrName, attributes[attrName]);
+        }
+        rect.setAttribute('stroke-width', shape.strokeWidth);
+        var rectTag = rect.outerHTML;
+        return this.isSVGTagValid(rectTag) ? rectTag : '';
```

Comment:
General question: Should you be throwing any error in case of non-validity? If not, how will the creator know?

Ditto elsewhere.

Reply:
These svg render functions are used internally by literallyCanvas and so the input and the output will be provided by literallyCanvas so I didn't want to break it by giving exception but just returning an empty string if the tag is invalid. This will not cause any problem to the creator because only the invalid tags are removed and the valid tags are used.

Reply:
But then wouldn't the creator be (unknowingly) exporting an image that is different to what they originally created?

Also, do we expect such issues to happen in practice / in normal usage? If not, it would be better to be noisy when an error occurs since that indicates some kind of problem. But if such errors may legitimately happen through normal operation then I suggest thinking more about what circumstances would lead to those and seeing if we can mitigate them.

Reply:
This issue is not expected to happen in normal usage at all. Now an error is thrown when the tag is not valid. Done.